File: LinkList/Remove_Nth_Node_From_End of_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {

    public ListNode reverse(ListNode head){
        ListNode cur =head,prev=null;
        while(cur!=null){
            ListNode temp=cur.next;
            cur.next=prev;
            prev=cur;
            cur=temp;
        }
        return prev;
    }

    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode s1=reverse(head);
        int k=n-1;
        if(k==0){
           s1=s1.next;
           return reverse(s1);
        }
        ListNode ab=s1;
        {
            for(int i=0;i<k-1;i++){
                ab=ab.next;
            }
            ab.next=ab.next.next;
            return reverse(s1);
        }
    }
}

File: LinkList/LinkList_cpp/RemoveNthNodeFromEndofList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *removeNthFromEnd(ListNode *head, int n)
    {
        if (head == nullptr)
        {
            return head;
        }
        ListNode *cur = head;
        int cnt = 0;
        while (cur != nullptr)
        {
            cnt++;
            cur = cur->next;
        }
        cur = head;
        for (int i = 0; i < cnt - n - 1; i++)
        {
            cur = cur->next;
        }
        if (cnt == n)
        {
            return head->next;
        }
        else
        {

            cur->next = cur->next->next;
        }
        return head;
    }
};